From e2eadc5e281c018092a497caf4ba535b7ad35edd Mon Sep 17 00:00:00 2001
From: Joe Hirschfeld <j@ibj.io>
Date: Sat, 2 Dec 2023 13:57:19 -0800
Subject: [PATCH 3/6] Fix inconsistencies in the generation of version.h

Current verison of echo does not correctly dump out two lines - printf will do the right thing
---
 Makefile | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 8d955d4..732dab3 100644
--- a/Makefile
+++ b/Makefile
@@ -74,14 +74,12 @@ endif
 
 .PHONY: version
 version:
-	echo "#define GIT_VERSION \"$(GIT_VERSION)\"\n#define GIT_REVISION \"$(GIT_REVISION)\"" > version.h.tmp
+	printf "#define GIT_VERSION \"$(GIT_VERSION)\"\n#define GIT_REVISION \"$(GIT_REVISION)\"\n" > version.h
 	if $(CXX) $(CFLAGS) -o /dev/null -c tests/libcamera/orientation.cc 2>/dev/null; then \
-		echo "#define LIBCAMERA_USES_ORIENTATION" >> version.h.tmp; \
+		printf "#define LIBCAMERA_USES_ORIENTATION\n" >> version.h; \
 	else \
-		echo "#define LIBCAMERA_USES_TRANSFORM" >> version.h.tmp; \
+		printf "#define LIBCAMERA_USES_TRANSFORM\n" >> version.h; \
 	fi
-	diff -u version.h version.h.tmp || mv version.h.tmp version.h
-	-rm -f version.h.tmp
 
 %: cmd/% $(TARGET_OBJS)
 	$(CCACHE) $(CXX) $(CFLAGS) -o $@ $(filter-out cmd/%, $^) $(filter $</%, $^) $(LDLIBS)
-- 
2.39.3 (Apple Git-145)

